src/components/App.js

```diff
@@ -27,43 +26,54 @@ export default class App extends Component {
   hideModal = () =>{
     this.setState({modalDisplayed: false});
   };
-  
+
   textChange = (e) => {
     this.setState({search: e.target.value})
-    this.filterSearch();
-    console.log(this.state.search);
+    this.beastFilter();
   };
 
-  filterSearch = () => {
+  updateHorns = (selection) => {
+    this.setState({hornFilter: selection});
+    console.log('horn state', this.state.hornFilter);
+  }
+  
+  beastFilter = () => {
     let beastArray = [];
+    let hornArray = [];
+    
+    switch (this.state.hornFilter) {
+      case "any":
+        hornArray = beastData;
+        break;
+      case "1":
+        hornArray = beastData.filter(arrObj => (arrObj.horns === 1));
+        break;
+      case "2":
+        hornArray = beastData.filter(arrObj => (arrObj.horns === 2));
+        break;
+      case "3":
+        hornArray = beastData.filter(arrObj => (arrObj.horns === 3));
+        break;  
+      default:
+        hornArray = beastData.filter(arrObj => (arrObj.horns > 3));
+        break;
+    }
```

Comment:
Really great job on the lab Erik!

I also like the switch case:) Another way to do this is to use one filter(). Inside we check if arrObj.horns === selection(the selection made by user), then update state. 👍

Reply:
Oh yea! A filter() would have been so much cleaner! I'm still getting used to having those tools in my toolbox! Thanks!